File: map-service/src/main/java/com/example/map/controller/CircleController.java
package com.example.map.controller;

import com.example.common.dto.ApiResponse;
import com.example.map.dto.CircleCreateRequest;
import com.example.map.dto.CircleResponse;
import com.example.map.service.CircleService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/map/groups")
@RequiredArgsConstructor
public class CircleController {

    private final CircleService circleService;

    @PostMapping
    public ApiResponse<CircleResponse> createCircle(
            @Valid @RequestBody CircleCreateRequest request,
            @RequestHeader("X-User-Id") Long userId) {
        CircleResponse response = circleService.createCircle(request, userId);
        return ApiResponse.success("그룹 생성 성공", response);
    }

    @GetMapping
    public ApiResponse<List<CircleResponse>> getMyCircles(
            @RequestHeader("X-User-Id") Long userId) {
        List<CircleResponse> circles = circleService.getMyCircles(userId);
        return ApiResponse.success("내 그룹 목록 조회 성공", circles);
    }
}

File: apps/map-service/src/circle/circle.controller.ts
import { Body, Controller, Get, Headers, Post } from '@nestjs/common';
import { ApiResponse } from '@app/common';
import { CircleService } from './circle.service';
import { CreateCircleDto } from './dto/create-circle.dto';
import { CircleResponseDto } from './dto/circle-response.dto';

@Controller('api/map/groups')
export class CircleController {
  constructor(private readonly circleService: CircleService) {}

  @Post()
  async createCircle(
    @Body() dto: CreateCircleDto,
    @Headers('x-user-id') userId: string,
  ): Promise<ApiResponse<CircleResponseDto>> {
    const response = await this.circleService.createCircle(dto, parseInt(userId));
    return ApiResponse.success('그룹 생성 성공', response);
  }

  @Get()
  async getMyCircles(
    @Headers('x-user-id') userId: string,
  ): Promise<ApiResponse<CircleResponseDto[]>> {
    const circles = await this.circleService.getMyCircles(parseInt(userId));
    return ApiResponse.success('내 그룹 목록 조회 성공', circles);
  }
}
